bundles/org.openhab.binding.juicenet/src/main/java/org/openhab/binding/juicenet/internal/api/JuiceNetApi.java

```diff
@@ -0,0 +1,217 @@
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.juicenet.internal.api.dto.JuiceNetApiDevice;
+import org.openhab.binding.juicenet.internal.api.dto.JuiceNetApiDeviceStatus;
+import org.openhab.binding.juicenet.internal.api.dto.JuiceNetApiInfo;
+import org.openhab.binding.juicenet.internal.api.dto.JuiceNetApiTouSchedule;
+import org.openhab.core.thing.ThingUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link JuiceNetApi} is responsible for implementing the api interface to the JuiceNet cloud server
+ *
+ * @author Jeff James - Initial contribution
+ */
+@NonNullByDefault
+public class JuiceNetApi {
+    private final Logger logger = LoggerFactory.getLogger(JuiceNetApi.class);
+
+    private static final String API_HOST = "https://jbv1-api.emotorwerks.com/";
+    private static final String API_ACCOUNT = API_HOST + "box_pin";
+    private static final String API_DEVICE = API_HOST + "box_api_secure";
+
+    private String apiToken = "";
+    private HttpClient httpClient;
+    private ThingUID bridgeUID;
+
+    public enum ApiCommand {
+        GET_ACCOUNT_UNITS("get_account_units", API_ACCOUNT),
+        GET_STATE("get_state", API_DEVICE),
+        SET_CHARGING_LIMIT("set_limit", API_DEVICE),
+        GET_SCHEDULE("get_schedule", API_DEVICE),
+        SET_SCHEDULE("set_schedule", API_DEVICE),
+        GET_INFO("get_info", API_DEVICE),
+        SET_OVERRIDE("set_override", API_DEVICE);
+
+        final String command;
+        final String uri;
+
+        ApiCommand(String command, String uri) {
+            this.command = command;
+            this.uri = uri;
+        }
+    }
+
+    public JuiceNetApi(HttpClient httpClient, ThingUID bridgeUID) {
+        this.bridgeUID = bridgeUID;
+        this.httpClient = httpClient;
+    }
+
+    public void setApiToken(String apiToken) {
+        this.apiToken = apiToken;
+    }
+
+    public List<JuiceNetApiDevice> queryDeviceList() throws JuiceNetApiException, InterruptedException {
+        JuiceNetApiDevice[] listDevices;
+        try {
+            JsonObject jsonResponse = postApiCommand(ApiCommand.GET_ACCOUNT_UNITS, null);
+
+            listDevices = new Gson().fromJson(jsonResponse.get("units").getAsJsonArray(), JuiceNetApiDevice[].class);
```

Comment:
`NullPointerException` possible - similar to https://github.com/openhab/openhab-addons/pull/10768#discussion_r1012297310.